apps/examples/iperf2/compat/Thread.c

```diff
@@ -0,0 +1,417 @@
+            Condition_Lock( thread_sNum_cond );
+            thread_sNum--;
+            Condition_Unlock( thread_sNum_cond );
+        }
+
+#elif defined( HAVE_WIN32_THREAD )
+
+        // Win32 threads -- spawn new thread
+        // Win32 has a thread handle in addition to the thread ID
+        thread->mHandle = CreateThread( NULL, 0, thread_run_wrapper, thread, 0, &thread->mTID );
+        if ( thread->mHandle == NULL ) {
+            WARN( 1, "CreateThread" );
+
+            // decrement thread count
+            Condition_Lock( thread_sNum_cond );
+            thread_sNum--;
+            Condition_Unlock( thread_sNum_cond );
+        }
+
+#else
+
+        // single-threaded -- call Run_Wrapper in this thread
+        thread_run_wrapper( thread );
+#endif
+    }
+} // end thread_start
+
+/* -------------------------------------------------------------------
+ * Stop the specified object's thread execution (if any) immediately.
+ * Decrements thread count and resets the thread ID.
+ * ------------------------------------------------------------------- */
+void thread_stop( struct thread_Settings* thread ) {
+
+#ifdef HAVE_THREAD
+    // Make sure we have been started
+    if ( ! thread_equalid( thread->mTID, thread_zeroid() ) ) {
+
+        // decrement thread count
+        Condition_Lock( thread_sNum_cond );
+        thread_sNum--;
+        Condition_Signal( &thread_sNum_cond );
+        Condition_Unlock( thread_sNum_cond );
+
+        // use exit()   if called from within this thread
+        // use cancel() if called from a different thread
+        if ( thread_equalid( thread_getid(), thread->mTID ) ) {
+
+            // Destroy the object
+            Settings_Destroy( thread );
+
+            // Exit
+#if   defined( HAVE_POSIX_THREAD )
+            pthread_exit( NULL );
+#else // Win32
+            CloseHandle( thread->mHandle );
+            ExitThread( 0 );
+#endif
+        } else {
+
+            // Cancel
+#if   defined( HAVE_POSIX_THREAD )
+            // Cray J90 doesn't have pthread_cancel; Iperf works okay without
+#ifdef HAVE_PTHREAD_CANCEL
+            pthread_cancel( thread->mTID );
+#endif
+#else // Win32
+            // this is a somewhat dangerous function; it's not
+            // suggested to Stop() threads a lot.
+            TerminateThread( thread->mHandle, 0 );
+#endif
+
+            // Destroy the object only after killing the thread
+            Settings_Destroy( thread );
+        }
+    }
+#endif
+} // end Stop
+
+/* -------------------------------------------------------------------
+ * This function is the entry point for new threads created in
+ * thread_start.
+ * ------------------------------------------------------------------- */
+#if   defined( HAVE_WIN32_THREAD )
+DWORD WINAPI
+#else
+void*
+#endif
+thread_run_wrapper( void* paramPtr ) {
+    struct thread_Settings* thread = (struct thread_Settings*) paramPtr;
+
+    // which type of object are we
+    switch ( thread->mThreadMode ) {
```

Comment:
coding rules: 'case' has a same indent with 'switch,'

Reply:
it's opensource so It would be better not to apply coding rule

Reply:
@pillip8282 If you want not to follow our coding rule, you should put it in external. If you don't, you should follow it. We negotiated only external because of what you say.